packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -64,7 +64,7 @@ module.exports = {
   // You can exclude the *.map files from the build during deployment.
   devtool: 'source-map',
   // In production, we only want to load the polyfills and the app code.
-  entry: [require.resolve('./polyfills'), paths.appIndexJs],
+  entry: [require.resolve('./polyfills'), paths.appIndexJs, paths.appVendors],
```

Comment:
Can you explain this to me? I thought it needed to be like this:
```js
{
  entry: {
    vendors: require(paths.appVendors),
    app: [require.resolve('./polyfills'), paths.appIndexJs]
  }
}
```

Reply:
I've done this to remain coherent with the current code
I think that 
```{
  entry: {
    vendors: require(paths.appVendors),
    app: paths.appIndexJs,
    polyfills: require.resolve('./polyfills'),
  }
}```
is easier to understand

Reply:
That seems fine as long as the polyfills get included before the app, and vendors before the app too.

I don't fully understand webpack entries 😄 .

Reply:
you can give either a string which indicates the location of a file (or a npm package), an array of those strings or an object.
If it's an array or a string, the name of the entry will be the name of the file, and if you give an object, the name is the key and the location is the value.

In this case, the vendors in an array so webpack's gonna concatenate the vendors into a single chunk called vendors.[hash].js and the app and the polyfills are linked to the location of the file

I haven't put the files in the right order, I'll change this

Reply:
the order of the entries is handle by HtmlWebpackPlugin with its option chunksSortMode

Reply:
> chunksSortMode

So do we need to turn this option on?
And are the entries intentionally in reverse order?

Reply:
I'm working on adding the option `chunksSortMode`
And the entries are now in a random order, I'll change this too (and also add chunk names for code splitting)

Reply:
Did you have a chance to check out the [article I linked](https://medium.com/webpack/predictable-long-term-caching-with-webpack-d3eee1d3fa31)? I'd like to know your thoughts on adding all of these plugins/options to achieve optimal caching.

Reply:
Most of what I've done is extracted from this article and from the origial issue #2206

Reply:
and from this article https://stackoverflow.com/questions/39899883/ordering-multiple-entry-points-in-webpack/46270611#46270611

Reply:
What's different between the current code on the MR and on the article is the `NamedChunksPlugin` that I haven't add yet and the plugin `NameAllModulesPlugin` but the doc says that it's recommanded to use in dev with HMR, so I'm not gonna add it

Reply:
Fair enough, but what I linked was about something else entirely -- we can address it in a different PR.

Reply:
Your article was about long term caching, and splitting the vendors into a separate file properly is a part of this

Reply:
Yeah, it is part. 😄  I was just seeing if you wanted to knock out the vendor bundle *and* caching in this PR -- but let's keep this PR focused.

Reply:
I think I've done both, but initially I only wanted to tackle the vendors but ag mention the article, I thought you wanted me to add this too :grin:

Reply:
Oh no it's totally fine if you did! 😄

Reply:
From what I understand, `NameAllModulesPlugin` is required for when externals are added because they cause ids to shift.

Sure, it may not affect non-ejected users, but we'd like ejected users to have something that "just works".

Reply:
the module `NameAllModulesPlugin` is deprecated and now we have to use `NamedModulesPlugin`

Reply:
(In fact, it's not, I have to learn how to read...)

Reply:
Haha you're alright! You're a total champ for your work on this PR; most people give up by now.

Reply:
Dude, it's just a two day work and I already did the split vendor thing in one of my project haha 😀